Torch.Server/Views/ModListControl.xaml.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using System.Runtime.CompilerServices;
+using System.Windows.Threading;
+using VRage.Game;
+using NLog;
+using Torch.Server.Managers;
+using Torch.API.Managers;
+using Torch.Server.ViewModels;
+using Torch.Server.Annotations;
+
+namespace Torch.Server.Views
+{
+    /// <summary>
+    /// Interaction logic for ModListControl.xaml
+    /// </summary>
+    public partial class ModListControl : UserControl, INotifyPropertyChanged
+    {
+        private static Logger Log = LogManager.GetLogger(nameof(ModListControl));
+        private InstanceManager _instanceManager;
+        ModItemInfo _draggedMod;
+        bool _hasOrderChanged = false;
+        bool _isSortedByLoadOrder = true;
+
+        //private List<BindingExpression> _bindingExpressions = new List<BindingExpression>();
+        /// <summary>
+        /// Constructor for ModListControl 
+        /// </summary>
+        public ModListControl()
+        {
+            InitializeComponent();
+            _instanceManager = TorchBase.Instance.Managers.GetManager<InstanceManager>();
+            _instanceManager.InstanceLoaded += _instanceManager_InstanceLoaded;
+            //var mods = _instanceManager.DedicatedConfig?.Mods;
+            //if( mods != null)
+            //    DataContext = new ObservableCollection<MyObjectBuilder_Checkpoint.ModItem>();
+            DataContext = _instanceManager.DedicatedConfig?.Mods;
+
+            // Gets called once all children are loaded
+            //Dispatcher.BeginInvoke(DispatcherPriority.Loaded, new Action(ApplyStyles));
+        }
+
+        private void ModListControl_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ResetSorting()
+        {
+            CollectionViewSource.GetDefaultView(ModList.ItemsSource).SortDescriptions.Clear();
+        }
+
+
+        private void _instanceManager_InstanceLoaded(ConfigDedicatedViewModel obj)
+        {
+            Log.Info("Instance loaded.");
+            //Dispatcher.Invoke(() => DataContext = new ObservableCollection<MyObjectBuilder_Checkpoint.ModItem>(obj.Mods));
+            Dispatcher.Invoke(() => DataContext = obj?.Mods ?? new ObservableCollection<ModItemInfo>());
```

Comment:
Combine these invokes into one to reduce allocations.

Reply:
Fixed in commit [cbbb196](https://github.com/TorchAPI/Torch/pull/263/commits/cbbb196d01c18eb587b09bd1d0104602e2f43bf1)